### PR TITLE
New emulator option

### PR DESCRIPTION
**Pull request by lauracabayol**
This branch includes a new emulator option for Nyx, which uses alphaP  in the parametrization of the P1D.
* Modify the archive to also import alphaP
* Modify the emulator to run with different number of input parameters depending on the version
* Estimate the normalization from all the archive, even if we drop some simulation for the leave one out test
* New emulator label: Nyx_alphap
* Pretrained models for Nyx_alphap

### CODE REVIEW

File: lace/utils/distance_to_center.py [hunk 1]
```diff
@@ -27,6 +27,12 @@ def calculate_distance_to_center(sim_test,
             archive = NyxArchive(verbose=False)
         else:
             raise ValueError("Available sim suites are 'mpg' and 'nyx'") 
+<<<<<<< HEAD
```
Comment (jchavesmontero) [inline lace/utils/distance_to_center.py:30]: There are conflicts here!

File: lace/emulator/nn_emulator.py [hunk 2]
```diff
@@ -457,7 +600,15 @@ def _obtain_sim_params(self):
             for i in range(len(self.training_data))
         ]
         training_label = np.array(training_label)
-        self.yscalings = np.median(training_label)
+                
+        if not self.emulator_label in ['Cabayol23', 'Cabayol23_extended']:
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:604]: Are computing the power spectrum summary statistics here? This should be done in the archives, in fact, we are already doing it.

File: lace/emulator/nn_emulator.py [hunk 3]
```diff
@@ -515,13 +665,20 @@ def _get_training_pd1_nn(self):
         ]
 
         training_label = np.array(training_label)
-        # yscalings = np.median(training_label)
-        training_label = np.log10(training_label / self.yscalings)
+        
+        if not self.emulator_label in ['Cabayol23', 'Cabayol23_extended']:
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:669]: Look at previous comment

File: lace/emulator/nn_emulator.py [hunk 4]
```diff
@@ -550,45 +708,68 @@ def train(self):
...
-        )  #
+            pretrained_model = torch.load(
+                os.path.join(self.models_dir, self.model_path),
+                map_location="cpu",
+            )
+            self.nn.load_state_dict(pretrained_model["emulator"])
+            self.nn.to(self.device)
+            self.print("Loading pretrained initial state.")
+
+        optimizer = optim.AdamW(
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:736]: Given all these changes, I guess that optimizing the previous emulators (Cabayol23) would lead to different results, right? It is not a problem, just something that we should take into account

**Review by jchavesmontero — CHANGES_REQUESTED** [review]
I left you a few comments, and there are conflicts in one of the files

File: lace/emulator/nn_emulator.py (hunk 2 above)
Comment (lauracabayol) [reply lace/emulator/nn_emulator.py:604]: This is for the normalization. It has always been inside the emulator code. This has also been useful to test different scalings and by doing it within the emulator, we make sure that the conversion back is correct. Do you have anything agains keeping this here?

File: lace/emulator/nn_emulator.py (hunk 4 above)
Comment (lauracabayol) [reply lace/emulator/nn_emulator.py:736]: Yes, sorry, this is something I wanted to make the distinction with the version in the paper and I forgot. Will add it.

File: lace/utils/distance_to_center.py (hunk 1 above)
Comment (lauracabayol) [reply lace/utils/distance_to_center.py]: fixed, thanks!

File: lace/emulator/nn_emulator.py (hunk 4 above)
Comment (lauracabayol) [reply lace/emulator/nn_emulator.py]: added! thanks!

File: lace/emulator/nn_emulator.py (hunk 2 above)
Comment (jchavesmontero) [reply lace/emulator/nn_emulator.py:604]: Ah, ok! I thought it was the summary statistics, but I can see now that it is the polynomial fit. Forget about it :)

**Comment by lauracabayol** [comment]
Thanks for the comments and suggestions!  @jchavesmontero,  please take a look at the  emulator manager file. I think it should be correct but I do not have a code to test it (that I know)
